lib/kb_fasttree/fasttree.py

```diff
@@ -0,0 +1,738 @@
+"""Implementation of the fasttree app."""
```

Comment:
The majority of this is a big ol' cut and paste from the impl file. I separated out a couple of functions to make it easier to mock certain things for testing. There's a whole load of cruft still in the code that I will deal with in an upcoming PR.

Reply:
I'll comment anything that I did change